scripts/git_changes_utils.py

```diff
@@ -93,6 +93,45 @@ def get_local_git_repository_remote_name() -> Optional[bytes]:
     return remote_name
 
 
+def get_local_git_repository_remote_name() -> Optional[bytes]:
```

Comment:
I just realized I don't actually understand what this means, especially when contrasted with the function above. What is the "remote name of the local repository"?

Also please don't copy-paste large chunks of code between functions. Extract the common bits if possible or pass in an argument if it makes sense. Otherwise it makes maintenance harder.

EDIT: Also, just to check, what's the difference between the two functions and does it make sense to update other references in the codebase to use the new function too?

Reply:
Hm, I kept the upstream one since the --run_on_changed_files_in_branch runs on all files in the current branch (don't know if this is useful or not) and I could just change it to only files that are being pushed. In the latter case I can just remove the upstream function and keep the local repository one for all cases.

Remote name of the local repository is the remote name for your fork of Oppia like jnvtnguyen/oppia.git instead of the "upstream", oppia/oppia.git. In the case that we only compare the files being pushed, we don't need the upstream one at all since we don't need to compare against develop anymore.

Reply:
Hm, I see. The only case I can think of when running on all files in the current branch would be useful is when a developer skipped the pre-push check in an earlier commit. But I don't really see them taking the trouble to manually add a flag for commits after that TBH. So I'm OK with dropping the upstream function and the extra flag, and removing all references to the flag, if you agree with that.

Ack on the remote name of the local repo, thanks.

Also did you see the first paragraph of my comment?

Reply:
Sorry, ack on the copy and paste; will keep it mind next time.
Removed the upstream function, but changed the run_on_changed_files_in_branch flags to use the local git repository; still think its useful if you want to test multiple times before pushing (instead of using git reset).